web-app/HeimScripts/heatmap/preprocess.R

```diff
@@ -0,0 +1,47 @@
+library("WGCNA")
+
+
+main <- function(aggregate=FALSE){
+  msgs = c("")
+  df <- loaded_variables[[length(loaded_variables)]]
+  good.input <- ncol(df) > 3
+  if(aggregate && good.input){
+    df <- dropEmptyGene(df)
+    aggr  <- aggregate.probes(df)
+    assign("preprocessed", aggr, envir = .GlobalEnv)
+    Discarded.rows <- nrow(df) - nrow(aggr)
+    msgs <- c( paste("Total discarded rows:",Discarded.rows))
+  }
+  else{
```

Comment:
The function main will only be called if aggregate = FALSE? 
Else, if "aggregate && good.input" is FALSE, the following message will also be produced when no probe aggregation is needed but the input is of the right format for probe aggregation:
msgs <- c("Incorrect subset - in order to perform probe aggregation more than one samples are needed.")